mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -429,6 +429,18 @@ def delete_run(self, run_id):
             run.lifecycle_stage = LifecycleStage.DELETED
             self._save_to_db(objs=run, session=session)
 
+    def _hard_delete_run(self, run_id):
+        with self.ManagedSessionMaker() as session:
+            run = self._get_run(run_uuid=run_id, session=session)
+            session.delete(run)
+
+    def _get_deleted_runs(self):
```

Comment:
Based on my comment above, I think we can remove this method and rely on the Search API instead.

Reply:
See my answer to the first comment about this.

Reply:
@t-henri That makes sense. This approach is also more efficient for the SqlAlchemyStore than `search()` because it only fetches run IDs.